### PR TITLE
feat(nextjs): init

### PR DESCRIPTION
**Pull request by Dandraghas**
## 🎉 Theme for [Next.js](https://nextjs.org) 🎉

The React Framework for the Web

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/src/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - `catppuccin.user.css` - all the CSS for the userstyle, based on the
    template.
  - `assets/mocha.webp`, `assets/macchiato.webp`, `assets/frappe.webp` and
    `assets/latte.webp` - individual screenshots of the themed website, in all 4
    Catppuccin flavors.
  - `catwalk.webp` - image of all individual screenshots stitched together,
    generated via [catwalk](https://github.com/catppuccin/toolbox#-catwalk).

### CODE REVIEW

File: scripts/userstyles.yml [hunk 1]
```diff
@@ -622,4 +625,11 @@ userstyles:
           answer: "It's available in case you have an OLED display. If you have one, you might want to enable this."
       current-maintainers: [*isabelroses]
       past-maintainers: [*elkrien]
+  nextjs:
+    name: Next.js
+    category: development
+    icon: nextjs
+    readme:
+      app-link: "https://nextjs.org"
+      current-maintainers: [ *Dandraghas ]
```
Comment (uncenter) [inline scripts/userstyles.yml:634]:
```suggestion
      current-maintainers: [*Dandraghas]
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Review by isabelroses — COMMENTED** [review]
Currently, links and other small things aren't themed see https://nextjs.org/docs for a set of examples.

**Comment by uncenter** [comment]
You did the gray colors but not the rest of them (i.e. blue for links as @isabelroses said):
```css
.dark-theme, .invert-theme, :host, :root {
	--ds-gray-100: hsla(var(--ds-gray-100-value),1);
	--ds-gray-200: hsla(var(--ds-gray-200-value),1);
	--ds-gray-300: hsla(var(--ds-gray-300-value),1);
	--ds-gray-400: hsla(var(--ds-gray-400-value),1);
	--ds-gray-500: hsla(var(--ds-gray-500-value),1);
	--ds-gray-600: hsla(var(--ds-gray-600-value),1);
	--ds-gray-700: hsla(var(--ds-gray-700-value),1);
	--ds-gray-800: hsla(var(--ds-gray-800-value),1);
	--ds-gray-900: hsla(var(--ds-gray-900-value),1);
	--ds-gray-1000: hsla(var(--ds-gray-1000-value),1);
	--ds-blue-100: hsla(var(--ds-blue-100-value),1);
	--ds-blue-200: hsla(var(--ds-blue-200-value),1);
	--ds-blue-300: hsla(var(--ds-blue-300-value),1);
	--ds-blue-400: hsla(var(--ds-blue-400-value),1);
	--ds-blue-500: hsla(var(--ds-blue-500-value),1);
	--ds-blue-600: hsla(var(--ds-blue-600-value),1);
	--ds-blue-700: hsla(var(--ds-blue-700-value),1);
	--ds-blue-800: hsla(var(--ds-blue-800-value),1);
	--ds-blue-900: hsla(var(--ds-blue-900-value),1);
	--ds-blue-1000: hsla(var(--ds-blue-1000-value),1);
	--ds-amber-100: hsla(var(--ds-amber-100-value),1);
	--ds-amber-200: hsla(var(--ds-amber-200-value),1);
	--ds-amber-300: hsla(var(--ds-amber-300-value),1);
	--ds-amber-400: hsla(var(--ds-amber-400-value),1);
	--ds-amber-500: hsla(var(--ds-amber-500-value),1);
	--ds-amber-600: hsla(var(--ds-amber-600-value),1);
	--ds-amber-700: hsla(var(--ds-amber-700-value),1);
	--ds-amber-800: hsla(var(--ds-amber-800-value),1);
	--ds-amber-900: hsla(var(--ds-amber-900-value),1);
	--ds-amber-1000: hsla(var(--ds-amber-1000-value),1);
	--ds-red-100: hsla(var(--ds-red-100-value),1);
	--ds-red-200: hsla(var(--ds-red-200-value),1);
	--ds-red-300: hsla(var(--ds-red-300-value),1);
	--ds-red-400: hsla(var(--ds-red-400-value),1);
	--ds-red-500: hsla(var(--ds-red-500-value),1);
	--ds-red-600: hsla(var(--ds-red-600-value),1);
	--ds-red-700: hsla(var(--ds-red-700-value),1);
	--ds-red-800: hsla(var(--ds-red-800-value),1);
	--ds-red-900: hsla(var(--ds-red-900-value),1);
	--ds-red-1000: hsla(var(--ds-red-1000-value),1);
	--ds-green-100: hsla(var(--ds-green-100-value),1);
	--ds-green-200: hsla(var(--ds-green-200-value),1);
	--ds-green-300: hsla(var(--ds-green-300-value),1);
	--ds-green-400: hsla(var(--ds-green-400-value),1);
	--ds-green-500: hsla(var(--ds-green-500-value),1);
	--ds-green-600: hsla(var(--ds-green-600-value),1);
	--ds-green-700: hsla(var(--ds-green-700-value),1);
	--ds-green-800: hsla(var(--ds-green-800-value),1);
	--ds-green-900: hsla(var(--ds-green-900-value),1);
	--ds-green-1000: hsla(var(--ds-green-1000-value),1);
	--ds-teal-100: hsla(var(--ds-teal-100-value),1);
	--ds-teal-200: hsla(var(--ds-teal-200-value),1);
	--ds-teal-300: hsla(var(--ds-teal-300-value),1);
	--ds-teal-400: hsla(var(--ds-teal-400-value),1);
	--ds-teal-500: hsla(var(--ds-teal-500-value),1);
	--ds-teal-600: hsla(var(--ds-teal-600-value),1);
	--ds-teal-700: hsla(var(--ds-teal-700-value),1);
	--ds-teal-800: hsla(var(--ds-teal-800-value),1);
	--ds-teal-900: hsla(var(--ds-teal-900-value),1);
	--ds-teal-1000: hsla(var(--ds-teal-1000-value),1);
	--ds-purple-100: hsla(var(--ds-purple-100-value),1);
	--ds-purple-200: hsla(var(--ds-purple-200-value),1);
	--ds-purple-300: hsla(var(--ds-purple-300-value),1);
	--ds-purple-400: hsla(var(--ds-purple-400-value),1);
	--ds-purple-500: hsla(var(--ds-purple-500-value),1);
	--ds-purple-600: hsla(var(--ds-purple-600-value),1);
	--ds-purple-700: hsla(var(--ds-purple-700-value),1);
	--ds-purple-800: hsla(var(--ds-purple-800-value),1);
	--ds-purple-900: hsla(var(--ds-purple-900-value),1);
	--ds-purple-1000: hsla(var(--ds-purple-1000-value),1);
	--ds-pink-100: hsla(var(--ds-pink-100-value),1);
	--ds-pink-200: hsla(var(--ds-pink-200-value),1);
	--ds-pink-300: hsla(var(--ds-pink-300-value),1);
	--ds-pink-400: hsla(var(--ds-pink-400-value),1);
	--ds-pink-500: hsla(var(--ds-pink-500-value),1);
	--ds-pink-600: hsla(var(--ds-pink-600-value),1);
	--ds-pink-700: hsla(var(--ds-pink-700-value),1);
	--ds-pink-800: hsla(var(--ds-pink-800-value),1);
	--ds-pink-900: hsla(var(--ds-pink-900-value),1);
	--ds-pink-1000: hsla(var(--ds-pink-1000-value),1);
}
```

**Comment by uncenter** [comment]
Also, the app and pages router icons are unthemed.
image

**Comment by uncenter** [comment]
It actually looks quite good, nice job. Placeholder text in search/input boxes is still unthemed (`input::placeholder`, see https://developer.mozilla.org/en-US/docs/Web/CSS/::placeholder).

File: styles/nextjs/catppuccin.user.css [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      --geist-stroke: var(--geist-background) !important;
+    }
+
+    button.navbar_search__dZT2b {
+      background-color: @mantle !important;
+    }
+
+    input::placeholder {
+      color: @text;
+    }
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:133]: Also themes the fake placeholder text in the search bar at the top of the page.
```suggestion
    input::placeholder,
    button[class*="navbar_search"] {
      color: @text;
    }
```

**Review by uncenter — CHANGES_REQUESTED** [review]
Looking good! A few small things I noticed...

**Social icons in footer:**
Screenshot 2024-02-05 at 22 21 57 (Arc)
These SVG icons can be themed, if you click on the element in the dev tools panel:
```html
<img alt="GitHub Logo" loading="lazy" width="16" height="16" decoding="async" data-nimg="1" style="color:transparent" src="/icons/github.svg">
```
You can see that it is an `` tag that links to an SVG. It's a little difficult to explain the process for theming these in a review comment, so I'll just give you the "answer" now and I promise we will document it properly in the future (see #449).

**Light mode:**
Without changing your Next.js theme, change your selected Stylus theme to Latte and observe the website. Looks like you missed a few spots:

Inline code, code block borders.
Screenshot 2024-02-05 at 22 41 24 (Arc)

Theme toggle shouldn't have a border around it.
Screenshot 2024-02-05 at 22 42 59 (Arc)

These cards look a little strange (compare Next.js dark mode + Latte to Next.js light mode).
Screenshot 2024-02-05 at 22 43 55 (Arc)

...and probably a few smaller things, I recommend that you try doing what I did, Latte with Next.js dark mode, to see what spots look off.

File: styles/nextjs/catppuccin.user.css [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      background: @base;
+    }
+
+    .gradient {
+      --text-gradient: @text;
+    }
+
+    .gradient-text {
+      --text-gradient: @text;
+    }
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:171]: Themes the two footer icons.

```suggestion
    .gradient-text {
      --text-gradient: @text;
    }
    
    img[alt="GitHub Logo"] {
      @svg: escape('<svg xmlns="http://www.w3.org/2000/svg" width="16" height="15.673" viewBox="0 0 16 15.673"><path fill="@{subtext0}" fill-rule="evenodd" d="M7.976 0C3.566 0 0 3.592 0 8.035c0 3.552 2.285 6.559 5.454 7.623 0.396 0.08 0.541 -0.173 0.541 -0.386 0 -0.186 -0.013 -0.825 -0.013 -1.49 -2.219 0.479 -2.681 -0.958 -2.681 -0.958 -0.357 -0.931 -0.885 -1.171 -0.885 -1.171 -0.726 -0.492 0.053 -0.492 0.053 -0.492 0.806 0.053 1.228 0.825 1.228 0.825 0.713 1.224 1.862 0.878 2.324 0.665 0.066 -0.519 0.277 -0.878 0.502 -1.078 -1.77 -0.186 -3.632 -0.878 -3.632 -3.965 0 -0.878 0.317 -1.596 0.819 -2.155 -0.079 -0.2 -0.357 -1.024 0.079 -2.129 0 0 0.673 -0.213 2.192 0.825a7.669 7.669 0 0 1 1.994 -0.266c0.673 0 1.36 0.093 1.994 0.266 1.519 -1.038 2.192 -0.825 2.192 -0.825 0.436 1.104 0.158 1.929 0.079 2.129 0.515 0.559 0.819 1.277 0.819 2.155 0 3.087 -1.862 3.765 -3.645 3.965 0.291 0.253 0.541 0.732 0.541 1.49 0 1.078 -0.013 1.942 -0.013 2.208 0 0.213 0.145 0.466 0.541 0.386 3.169 -1.064 5.454 -4.071 5.454 -7.623C15.952 3.592 12.374 0 7.976 0z" clip-rule="evenodd"/></svg>');
      height: 0 !important;
      width: 0 !important;
      padding-left: 16px !important;
      padding-top: 16px !important;
      background: url("data:image/svg+xml,@{svg}") no-repeat !important;
    }
    img[alt="X Logo"] {
      @svg: escape('<svg xmlns="http://www.w3.org/2000/svg" fill="none" height="16" shape-rendering="geometricPrecision" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" viewBox="0 0 24 24" width="16"><path fill="@{subtext0}" d="M16.99 0h3.308l-7.227 8.26 8.502 11.24h-6.657l-5.214-6.817L3.736 19.5H.426l7.73-8.835L0 0h6.826l4.713 6.231L16.99 0Zm-1.161 17.52h1.833L5.83 1.876H3.863L15.829 17.52Z" stroke="none"/></svg>');
      height: 0 !important;
      width: 0 !important;
      padding-left: 16px !important;
      padding-top: 16px !important;
      background: url("data:image/svg+xml,@{svg}") no-repeat !important;
    }
```

**Comment by Dandraghas** [comment]
> Without changing your Next.js theme, change your selected Stylus theme to Latte and observe the website. Looks like you missed a few spots:
>
> Inline code, code block borders.

I am not sure why it looks like that for you

my screenshot
image
your screenshot:
image

same for cards on latte:
image

your screenshot:
image

I don't have any borders on inline code blocks, nor shadows on cards like you have

i mainly use firefox so i tried chromium, but i still unable to reproduce what you have

File: styles/nextjs/catppuccin.user.css [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+      background: @base;
+    }
+
+    .foundation_root__wm0ez {
+      background: @base;
+    }
+
+    .foundation_card__v7VKB {
+      background: @base;
+    }
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:177]:
```suggestion
    .features_card__xYla9,
    .foundation_root__wm0ez,
    .foundation_card__v7VKB {
      background: @base;
    }
```

File: styles/nextjs/catppuccin.user.css [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+    .gradient {
+      --text-gradient: @text;
+    }
+
+    .gradient-text {
+      --text-gradient: @text;
+    }
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:193]:
```suggestion
    .gradient,
    .gradient-text {
      --text-gradient: @text;
    }
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Comment by uncenter** [comment]
The https://nextjs.org/learn page has a few unthemed elements, most of it is fine though. Just scroll through different parts and toggle the userstyle on and off and see what needs to be changed.

Also, `--ds-gray-700` needs to be themed for the docs breadcrumbs:

Screenshot 2024-02-07 at 12 54 22 (Arc)

File: styles/nextjs/catppuccin.user.css [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+    [data-docs-table-of-contents] code:not(pre code) {
+      border-color: 1px solid @surface0 !important;
+      background-color: @mantle !important;
+    }
+
+    td {
+      a {
+        color: @accent-color;
+      }
+    }
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:169]:
```suggestion
    td a {
      color: @accent-color;
    }
```

**Review by uncenter — CHANGES_REQUESTED** [review]
Almost done! The hover color of the icons on these cards on the Learn page are unthemed. 
Screenshot 2024-02-07 at 16 36 09 (Arc)

**Review by uncenter — CHANGES_REQUESTED** [review]

File: styles/nextjs/catppuccin.user.css [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+      background: @base;
+    }
+
+    .foundation_cpu__ciXpm {
+      background: @mantle;
+    }
+
+    .header_header__zJOD0 {
+      background: @base;
+    }
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:193]: The following suggestion themes the mobile/small screen nav menu:
Screenshot 2024-02-08 at 15 44 08 (Arc)

```suggestion
    .header_header__zJOD0 {
      background: @base;
    }
    .mute, .subtitle {
      color: @subtext0;
    }

    .mobile-menu_mobileMenu__xqbOP a.mobile-menu_selected__G55UL {
      color: var(--geist-background);
    }

    .dark-theme .dark\:bg-black\/50 {
      background-color: var(--geist-background);
    }
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Review by uncenter — APPROVED** [review]

**Comment by uncenter** [comment]
Syntax highlighting isn't done yet but I'll figure that out for you, I'm pretty familiar with what they are using for syntax highlighting and it won't take long but I haven't had time to do it yet. I think it is fine to merge for now and I'll do a PR later.

File: styles/nextjs/catppuccin.user.css [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+
+    .bg-gray-50 {
+      background-color: @mantle;
+    }
+
+    .styled-scrollbar {
+      scrollbar-color: @accent-color transparent;
+    }
+
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:233]: Code block syntax highlighting!
```suggestion

      .code-block_pre___OLfy {
        --shiki-token-punctuation: @overlay2;
        --shiki-token-constant: @peach;
        --shiki-token-string: @green;
        --shiki-token-string-expression: @green;
        --shiki-token-comment: @overlay0;
        --shiki-token-keyword: @mauve;
        --shiki-token-parameter: @maroon;
        --shiki-token-function: @blue;
      }

```

File: styles/nextjs/catppuccin.user.css [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+    --geist-error: @red;
+    --geist-cyan: @teal;
+
+    --ds-background-100: @base;
+    --ds-background-200: @surface0;
+    --ds-gray-100: @base;
+    --ds-gray-200: @surface0;
+    --ds-gray-300: @surface1;
+    --ds-gray-400: @surface0;
+    --ds-gray-700: @subtext0;
```
Comment (uncenter) [inline styles/nextjs/catppuccin.user.css:84]: Line numbers in code blocks.
```suggestion
    --ds-gray-400: @surface0;
    --ds-gray-600: @overlay1;
    --ds-gray-700: @subtext0;
```

**Review by uncenter — APPROVED** [review]

**Review by isabelroses — APPROVED** [review]
Thanks for the PR and working along @uncenter suggestions.

**Comment by uncenter** [comment]
> Thanks for the PR and working along @uncenter suggestions.

Yeah sorry for all my nitpicks :sob:

**Pull request merged**